### PR TITLE
K8s updates from KTI

### PR DESCRIPTION
**Pull request by kti-richard**
This is based off a rebase from master, so recent work you have merged into master related to k8s will be in here.
The changes for appVersion are not though, your PR would change those.   I will review/test that later - I wanted to get the PR out.   There is an edge case for the url_overrides I wanted to clean up, but I will make a separate PR after that.
This has been tested on local standup with default values.yaml, and user creation via API.

### CODE REVIEW

**Comment by yonabuild** [comment]
Can one of the admins verify this patch?

**Comment by kti-richard** [comment]
The hope will be to move to smaller PR going forward :)

**Comment by Bert-R** [comment]
add to whitelist

File: k8s/helm/yona-server/values.yaml [hunk 1]
```diff
@@ -166,10 +134,12 @@ sms:
   
 apple:
   enabled: false
+  keystore_secret_filename: dummy.p12
```
Comment (Bert-R) [inline k8s/helm/yona-server/values.yaml]: Why are you making the file name for the secret a variable? The name for the keystore itself remains "dummy".
I would rather move away from dummy in all scenarios and just vary the content between "real" and "dummy" while the name remains serious.

File: k8s/helm/yona-server/templates/01_configmap_properties.yaml [hunk 2]
```diff
@@ -44,14 +45,15 @@ data:
     yona.ldap.accessUserDn={{ .Values.ldap.user_dn | default "cn=admin,dc=example,dc=local" }}
     yona.ldap.accessUserPassword={{ .Values.ldap.user_password | default "password" }}
     yona.sms.enabled= {{ .Values.sms.enabled | default "false" }}
-    yona.sms.defaultSenderNumber = {{ .Values.sms.default_sender_number | default "+17184169858" }}
+    yona.sms.defaultSenderNumber = {{ .Values.sms.default_sender_number | default "+12312312312" }}
     yona.sms.alphaSenderId = {{ .Values.sms.alpha_sender_id | default "Yona" }}
-    yona.sms.alphaSenderSupportingCountryCallingCodes = {{ .Values.sms.alpha_sender_supporting_country_calling_codes | default "+31 +49 +33" }}
+    yona.sms.alphaSenderSupportingCountryCallingCodes = {{ .Values.sms.alpha_sender_supporting_country_calling_codes | default "+31 +49 +33" | quote}}
```
Comment (kti-richard) [inline k8s/helm/yona-server/templates/01_configmap_properties.yaml]: Ensure that the value is wrapped in quotes if needed.
Habbit - Needed for yaml generated, but not needed for raw application.properties generation - PR ammended to remove.

File: k8s/helm/yona-server/templates/10_deployment_app.yaml [hunk 3]
```diff
@@ -38,17 +38,11 @@ spec:
               value: "jdbc:mariadb://{{ .Release.Name }}-mariadb.{{ .Release.Namespace }}.svc.cluster.local/{{ .Values.mariadb.mariadbDatabase }}"
             {{- end }}
             - name: YONA_ANALYSIS_SERVICE_SERVICE_URL
-            {{- if .Values.analysis }}
+            {{- if hasKey .Values.analysis "url_override" }}
               value: {{ .Values.analysis.url_override | quote }}
             {{- else }}
               value: "http://analysis.{{ .Release.Namespace }}.svc.cluster.local"
             {{- end }}
-            - name: YONA_BATCH_SERVICE_SERVICE_URL
```
Comment (kti-richard) [inline k8s/helm/yona-server/templates/10_deployment_app.yaml]: Bad merge, and part of the "edge cases" I was working on.    I have added the full one back and tested against prod and stock values.   I will still have to test with the hasKey handling for edge cases on missing values.yaml items.

File: k8s/helm/yona-server/values.yaml (hunk 1 above)
Comment (kti-richard) [inline k8s/helm/yona-server/values.yaml]: It was just something different between prod and default that I had to alter on that day.

The filename is not dummy on prod, but I didn't want to keep default values to generate as close to the same reference material in resources/*

Making it a variable allowed me to set the production value to the desired production name.  We can make it all the same as on production, just need to change values.yaml.    I have not altered the PR to reflect this though.

File: k8s/helm/yona-server/templates/01_configmap_properties.yaml [hunk 4]
```diff
@@ -44,14 +45,15 @@ data:
     yona.ldap.accessUserDn={{ .Values.ldap.user_dn | default "cn=admin,dc=example,dc=local" }}
     yona.ldap.accessUserPassword={{ .Values.ldap.user_password | default "password" }}
     yona.sms.enabled= {{ .Values.sms.enabled | default "false" }}
-    yona.sms.defaultSenderNumber = {{ .Values.sms.default_sender_number | default "+17184169858" }}
+    yona.sms.defaultSenderNumber = {{ .Values.sms.default_sender_number | default "+12312312312" }}
```
Comment (kti-richard) [inline k8s/helm/yona-server/templates/01_configmap_properties.yaml]: Just a habbit of keeping unique production values out.  
Probably not of value, but I don't know - SMS #'s can be spoofed, etc.
I am not going to fight for it ;) I can revert.

File: k8s/README.md [hunk 5]
```diff
@@ -6,6 +6,14 @@ The deployment uses [Helm](https://github.com/kubernetes/helm) to template the r
 Locally we recommend the use of [MiniKube](https://kubernetes.io/docs/getting-started-guides/minikube/) for a local workstation setup of a single node Kubernetes environment.   Minikube will create a node using various virtualization technologies.  
 
 ----------
+## Windows Setup
```
Comment (kti-richard) [inline k8s/README.md]: Updated and added link to wiki

File: k8s/helm/yona-server/templates/01_configmap_properties.yaml (hunk 4 above)
Comment (Bert-R) [inline k8s/helm/yona-server/templates/01_configmap_properties.yaml]: We need to do something with the multitude of defaults. Let's make this part of that.

**Comment by Bert-R** [comment]
Looks good. I'm going to merge it.
